### PR TITLE
rewrite attach_dir_desc

### PR DESCRIPTION
**Pull request by AngelinaBelle**
Rewrite as per https://github.com/SimpleMachines/SMF2.1/issues/288
<br/> I agree to the DCO 1.1 at https://github.com/SimpleMachines/SMF2.1/commit/b486a1e52a5f4f1b9ffae37defc6cd864926267d#DCO.txt

### CODE REVIEW

File: Themes/default/languages/Admin.english.php [hunk 1]
```diff
@@ -385,7 +385,7 @@
...
 $txt['attach_dir_basedir'] = 'Base directory';
 
-$txt['attach_dir_desc'] = 'Create new directories or change the current directory below. Directories can be renamed as long as they do not contain a sub-directory. If the new directory is to be in the created within the forum directory structure. Just the directory name can be used. To remove a directory, blank the path input field. Directories can not be deleted if they contain either files or sub-directories (shown in brackets next to the file count).';
+$txt['attach_dir_desc'] = 'Create new directories or change the current directory below. <br />To create a new directory within the forum directory structure, use just the directory name. <br />To remove a directory, blank the path input field. Only empty directories can be removed. To see if a directory is empty, check for files or sub-directories in brackets next to the file count. <br /> To rename a directory, simply change its name in the input field. Only directories without sub-directories may be renamed.'Directories can be renamed as long as they do not contain a sub-directory. If the new directory is to be in the created within the forum directory structure.';
```
Comment (XinYenFon) [inline Themes/default/languages/Admin.english.php]: you have to escape quotes with backslash :)

Comment (AngelinaBelle) [inline Themes/default/languages/Admin.english.php]: You are correct.  I am working too fast and not testing my work, or even reading carefully.

Would it make sense for me to resubmit my pull request?

Comment (XinYenFon) [inline Themes/default/languages/Admin.english.php]: You don't have to resubmit just correct the mistake and push to your repo and github updates this PR for you.

Can you please also look at ManageSettings.english > $txt['enable_disregard'] there is small typo it looks I don't want to open new PR for such small change.

**Comment by ChainyEO** [comment]
The new version of "attach_dir_desc" also contains errors, so this needs reopening. Is anyone still watching this topic, or should I post somewhere else about it? Sorry, I've never used this system before.

**Comment by emanuele45** [comment]
Hi Chainy!

Ah..I see you mean the last bit, right?
I think it is enough to remove it, am I correct?

**Comment by emanuele45** [comment]
[This](https://github.com/emanuele45/playpen/commit/1715898ce226374edb5efab2b72e85e45718619f) should fix it.
Let me know if is fine and I'll merge the changes in the main repository. ;)

**Comment by ChainyEO** [comment]
Here are my thoughts on the above string, followed by a suggested re-wording.

Some parts of the string don't make sense to me. I set up a local installation of SMF2.1 and I took a look at the page in the administration center where this string is used (Administration Center > Attachments and Avatars > Attachment Directories). Consider the following:

"To create a new directory within the forum directory structure, use just the directory name."
- What does that mean? I suppose it means that the new directory should be created within the existing directory called "attachments". By the way, I tried creating a new directory via the administration center of the forum, but it didn't work. I ended up going directly to the directory and manually creating a new one there. 

"To see if a directory is empty, check for files or sub-directories in brackets next to the file count."
- I can't see what this is referring to. (see the attachment below) There are no brackets anywhere. I think this sentence can simply be left out. The administration center shows this information very clearly, so people don't need any extra pointers.

"Only directories without sub-directories may be renamed. Directories can be renamed as long as they do not contain a sub-directory."
- Two sentences, both of which say exactly the same thing.

"If the new directory is to be in the created within the forum directory structure."
- Emanuele, you rightly deleted this sentence as it makes no sense at all! 

How about the following re-wording (omitting the comments in brackets, of course):

"On this page you can create new directories for attachments, or change the directory that is currently used. 

To create a new directory within the existing directory structure of the forum, just enter the new directory name in a new field. (this still doesn't make much sense!)

To remove a directory, blank out the path input field. It is only possible to remove a directory if it contains no files or subdirectories. (this is clearer than just saying 'empty').

To rename a directory, simply change its name in the input field. Only directories without sub-directories may be renamed."

Screenshot - 040313 - 17:14:54

**Comment by emanuele45** [comment]
> "To create a new directory within the forum directory structure, use just the directory name."
> - What does that mean? I suppose it means that the new directory should be created within the existing directory called "attachments". By the way, I tried creating a new directory via the administration center of the forum, but it didn't work. I ended up going directly to the directory and manually creating a new one there.

It seems the intention is that one but you are right, it doesn't create the directory (and the error message seems wrong either), but just to be sure, let's see if @Kays48 has an idea. :angel:

**Comment by Kays48** [comment]
Sorry about the confusing wording. If you just enter a name into the field. A folder with that name should be created in the forum root. At the same level as the Attachments directory.

**Comment by emanuele45** [comment]
Then it seems we have a bugz! :D

I don't know when it has been broken, though if I put only the name of a dir I get the message that "the directory has been created, but is not writable" but there is not directory...

**Comment by ChainyEO** [comment]
Yes, same problem here. No directory was actually created, despite the warning message:

Screenshot - 040313 - 23:54:19

**Comment by ChainyEO** [comment]
By the way, I know it's become rather fashionable these days, but is it really necessary for us to adopt the rather annoying style of speech as demonstrated by "Oops, there seems to be a problem." (!?) The same goes for the irritating title to all the pop-up help messages "A little lost? Let me explain:" A user is clearly a little lost if he/she has just clicked on a question mark icon, so the question seems rather pointless. 

I suppose this is a question for another place, but I just thought I'd mention it...

**Comment by Kays48** [comment]
Yes, something is probably broken. I checked with my old repo and it works fine there.

Does it work properly if you enter in the  full path?

**Comment by ChainyEO** [comment]
Yes, if you put the full path in then it works. I just created new directories in the main smf root folder, and also a new one within the 'attachments' directory. Both times entering the full path.

**Comment by emanuele45** [comment]
Yes, the full path works fine.

I was wondering: this new directory is always created in $boarddir even if it is configured to create automatic directories inside another one?

**Comment by Kays48** [comment]
It is suppose to create it in $boarddir since it's a relative path. Same level as index.php

Hnm, I just did a fresh install of SMF2.1-release-2.1 and that works fine for me also. So I don't know if something is broken or if this is a matter of it not working in all server configurations.

Maybe just remove or change that line. Since if someone does just enter the directory name it's not guaranteed to work.

**Pull request closed without merging**